ui/src/components/WorkerManagerWorkerTypesTable/index.jsx

```diff
@@ -0,0 +1,154 @@
+    searchTerm: string,
+    path: string,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  sortWorkerTypes = memoize(
+    (workerTypes, sortBy, sortDirection, searchTerm) => {
+      const sortByProperty = camelCase(sortBy);
+      const filteredWorkerTypes = searchTerm
+        ? workerTypes.filter(({ workerType }) =>
+            workerType.includes(searchTerm)
+          )
+        : workerTypes;
+
+      return isEmpty(filteredWorkerTypes)
+        ? filteredWorkerTypes
+        : [...filteredWorkerTypes].sort((a, b) => {
+            const firstElement =
+              sortDirection === 'desc' ? b[sortByProperty] : a[sortByProperty];
+            const secondElement =
+              sortDirection === 'desc' ? a[sortByProperty] : b[sortByProperty];
+
+            return sort(firstElement, secondElement);
+          });
+    },
+    {
+      serializer: ([workerTypes, sortBy, sortDirection, searchTerm]) => {
+        const ids = pipe(
+          rSort((a, b) => sort(a.workerType, b.workerType)),
+          map(({ workerType }) => workerType)
+        )(workerTypes);
+
+        return `${ids.join('-')}-${sortBy}-${sortDirection}-${searchTerm}`;
+      },
+    }
+  );
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  renderRow = workerType => {
+    const { path } = this.props;
+    const iconSize = 16;
+    const { failedNumber, exceptionNumber, unscheduledNumber } = workerType;
+
+    return (
+      <TableRow key={workerType.name}>
+        <TableCell
+          button
+          component={Link}
+          to={`${path}/worker-types/${workerType.name}`}>
+          <Typography>{workerType.name}</Typography>
+          <LinkIcon size={iconSize} />
+        </TableCell>
+
+        <TableCell>
+          <Typography>{workerType.pendingCapacity}</Typography>
+        </TableCell>
+
+        <TableCell>
+          <Typography>{workerType.runningCapacity}</Typography>
+        </TableCell>
+
+        <TableCell>
+          <Typography>{workerType.pendingTasks}</Typography>
+        </TableCell>
+
+        <TableCell>
+          <Typography>
+            <DateDistance from={workerType.lastActive} />
+          </Typography>
+        </TableCell>
+
+        <TableCell>
+          <Typography>
+            <DateDistance from={workerType.lastResolved} />
+          </Typography>
+        </TableCell>
+
+        <TableCell>
+          <Typography>
+            {`${failedNumber} / ${exceptionNumber} / ${unscheduledNumber}`}
+          </Typography>
+        </TableCell>
+
+        <TableCell
+          button
+          component={Link}
+          to={`${path}/providers/${workerType.provider}`}>
+          <Typography>{workerType.provider}</Typography>
+          <LinkIcon size={iconSize} />
+        </TableCell>
```

Comment:
See comment above regarding link icon & hover state.

Reply:
which comment is it, can you link? I just went through all the comments above and didn't see the one about hover state. There are so many comments here, I am probably missing something

Reply:
https://github.com/taskcluster/taskcluster/pull/566#discussion_r277284935

Reply:
this suggestion is now implemented in both places